erpnext/stock/get_item_details.py

```diff
@@ -402,9 +402,17 @@ def get_basic_details(args, item, overwrite_warehouse=True):
 	):
 		from erpnext.buying.doctype.purchase_order.purchase_order import item_last_purchase_rate
 
-		out.last_purchase_rate = item_last_purchase_rate(
```

Comment:
`@request_cache` is a simpler solution to this

Reply:
https://github.com/frappe/frappe/blob/455c68ac553c135ec897491a1260672bfdf32fc4/frappe/utils/caching.py#L24